fitnessllm_dataplatform/task_handler.py

```
@@ -0,0 +1,63 @@
+            KeyError: If required options or environment variables are missing.
+            ValueError: If data source is not supported.
+        """
+        if 'data_source' not in options:
+            raise KeyError("Missing required option: data_source")
+
+        try:
+            InfrastructureNames = DynamicEnum.from_dict(
+                get_secret(environ["INFRASTRUCTURE_SECRET"])[environ["STAGE"]],
+            )
+        except KeyError as e:
+            raise KeyError(f"Missing environment variable: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)